app/models/minion.rb

```diff
@@ -1,4 +1,25 @@
 # frozen_string_literal: true
+
+require "pharos/salt_minion"
+
 # Minion represents the minions that have been registered in this application.
 class Minion < ApplicationRecord
+  enum role: [:master, :minion]
+
+  # rubocop:disable SkipsModelValidations
+  # Assigns a role to this minion locally in the database, and send that role
+  # to salt subsystem.
+  def assign_role(role:)
+    return false if salt.roles?
+    Minion.transaction do
+      update_column :role, role
+      salt.assign_role role: role
+    end
+    true
+  end
+
+  # Returns the proxy for the salt minion
+  def salt
+    @salt ||= Pharos::SaltMinion.new mid: hostname
```

Comment:
[MINOR] Maybe we don't need a keyword argument for this one but if we use one I suggest we rename it to minion_id. It's more verbose but it might not be obvious to someone as "mid" because after all we set it to "hostname".